specifyweb/frontend/js_src/lib/components/Merging/index.tsx

```diff
@@ -255,7 +268,7 @@ function Merging({
                 errorMode: 'dismissible',
               }
             ).then((response) => {
-              setMergeId(response.data);
+              setMergeId(JSON.parse(response.data));
               if (response.status === Http.NOT_ALLOWED) {
```

Comment:
refactor to use if/else instead of if+return - will be cleaner and less bug-prone that way

Reply:
how about instead of checking for Http.NOT_ALLOWED you check for everything that isn't what you expect - this way all errors will be caught

thus either `response.status !== Http.<what is the expected response code?>`
OR simply `!response.response.ok` (see https://developer.mozilla.org/en-US/docs/Web/API/Response/ok)

and use object destucturing to remove the need for weird looking `response.response` :
instead of ` ).then((response) => {` do ` ).then(({data,response}) => {`